pkg/analysis/passes/safelinks/safelinks.go

```diff
@@ -0,0 +1,164 @@
+package safelinks
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"net/url"
+	"os"
+	"strings"
+	"time"
+
+	"github.com/grafana/plugin-validator/pkg/analysis"
+	"github.com/grafana/plugin-validator/pkg/analysis/passes/metadata"
+)
+
+type LinkResult struct {
+	Link    metadata.Link
+	Threats []ThreatType
+	Error   error
+}
+
+type WebRiskResponse struct {
+	Threat *struct {
+		ThreatTypes []string `json:"threatTypes"`
+	} `json:"threat,omitempty"`
+}
+
+type ThreatType string
+
+const (
+	ThreatTypeMalware                           ThreatType = "MALWARE"
+	ThreatTypeSocialEngineering                 ThreatType = "SOCIAL_ENGINEERING"
+	ThreatTypeUnwantedSoftware                  ThreatType = "UNWANTED_SOFTWARE"
+	ThreatTypeSocialEngineeringExtendedCoverage ThreatType = "SOCIAL_ENGINEERING_EXTENDED_COVERAGE"
+)
+
+var webriskApiKey = os.Getenv("WEBRISK_API_KEY")
+
+var (
+	webriskFlagged    = &analysis.Rule{Name: "webrisk-flagged", Severity: analysis.Error}
+	webRiskAPIBaseURL = "https://webrisk.googleapis.com/v1/uris:search"
+	requestTimeout    = 15 * time.Second
+	httpClient        = &http.Client{}
+)
+
+var Analyzer = &analysis.Analyzer{
+	Name:     "safelinks",
+	Requires: []*analysis.Analyzer{metadata.Analyzer},
+	Run:      run,
+	Rules:    []*analysis.Rule{webriskFlagged},
+	ReadmeInfo: analysis.ReadmeInfo{
+		Name:        "Safe Links",
+		Description: "Checks that links from `plugin.json` are safe.",
+	},
+}
+
+func run(pass *analysis.Pass) (interface{}, error) {
+	metadataBody, ok := pass.ResultOf[metadata.Analyzer].([]byte)
+	if !ok {
+		return nil, nil
+	}
+
+	var data metadata.Metadata
+	if err := json.Unmarshal(metadataBody, &data); err != nil {
+		return nil, err
+	}
+
+	if len(data.Info.Links) == 0 {
+		return nil, nil
+	}
+
+	if webriskApiKey == "" {
+		return nil, nil
+	}
+
+	ctx := context.Background()
+
+	results := CheckURLs(ctx, data.Info.Links)
+
+	for _, result := range results {
+		if result.Error != nil || len(result.Threats) > 0 {
+			pass.ReportResult(pass.AnalyzerName, webriskFlagged,
+				"Webrisk flagged link",
+				fmt.Sprintf("Link with name %s is not safe: can be a %s", result.Link.Name, getThreatTypeString(result.Threats)))
+		}
+	}
+	return nil, nil
+}
+
+func CheckURLs(ctx context.Context, links []metadata.Link) []LinkResult {
+	results := make([]LinkResult, len(links))
+
+	for i, link := range links {
+		result := LinkResult{Link: link}
+
+		if link.URL == "" {
+			results[i] = result
+			continue
+		}
+
+		params := url.Values{}
+		params.Set("uri", link.URL)
+		params.Set("key", webriskApiKey)
```

Comment:
I don't like the idea of us sending the key over the http parameters of a get request. this can easily be logged in many places.

Check if we can use the header `X-Goog-Api-Key` instead as documented here https://cloud.google.com/docs/authentication/api-keys-use#using-with-rest